Lib/glyphsLib/builder/sources.py

```diff
@@ -53,6 +57,21 @@ def _to_designspace_source(self, master, is_regular):
         source.filename = build_ufo_path(
             '', source.familyName, source.styleName)
 
+        # Make sure UFO filenames are unique, lest we overwrite masters that
+        # happen to have the same weight name.
+        n = "_"
+        while any(s != source and s.filename == source.filename
```

Comment:
please replace `s != source` with `s is not source`

Reply:
I wonder if instead of a while loop called each time `for master in self.font.masters` (O(n^2)), we could store the set of filenames used so far in a private attribute of self and use that to check if each new master is given a unique filename..
But I guess the number of masters is unlikely to be huge.

Reply:
> please replace s != source with s is not source

Argh forgot about that.

> I wonder if instead of a while loop called each time

This would have to be benchmarked... I'd say leave for now.